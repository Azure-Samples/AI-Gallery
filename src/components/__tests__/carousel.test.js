import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Carousel from './../carousel';

configure({ adapter: new Adapter() });

test('renders a snapshot of SliderContent (models)', () => {
    const tree = renderer.create(<Carousel keyword={"models"}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('renders a snapshot of SliderContent (examples)', () => {
    const tree = renderer.create(<Carousel keyword={"examples"}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('updateSize returns the correct value', () => {
    const wrapper = shallow(<Carousel keyword={"models"}/>);
    expect(wrapper.instance().getMenuSize(100,100)).toBe(10000);
})

test('getScrollAmount returns correct value when provided Left', () => {
    const wrapper = shallow(<Carousel keyword={"models"}/>);
    expect(wrapper.instance().getScrollAmount(200, 500, 1200, "left")).toBe(0);
    expect(wrapper.instance().getScrollAmount(1000, 800, 1200, "left")).toBe(600);
})

test('scroll determines the correct paddles to show', () => {
    const wrapper = shallow(<Carousel keyword={"models"}/>);
    expect(wrapper.instance().scroll(0, 100, 900, 800)).toEqual([0, 1]);
    expect(wrapper.instance().scroll(400, 200, 900, 700)).toEqual([1, 1]);
    expect(wrapper.instance().scroll(300, 300, 600, 300)).toEqual([1, 0]);
    expect(wrapper.instance().scroll(0, 300, 200, 0)).toEqual([0,0]);
})