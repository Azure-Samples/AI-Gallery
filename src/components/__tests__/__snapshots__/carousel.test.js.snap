// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of SliderContent (examples) 1`] = `
<div>
  <ul
    className="menu examples"
    id="examples-menu"
  >
    <li
      className="examplesitem item"
    >
      <div
        class="github-card"
        data-github="Azure/MachineLearningSamples-PredictiveMaintenance"
        data-theme="medium"
      />
    </li>
    <li
      className="examplesitem item"
    >
      <div
        class="github-card"
        data-github="Azure/MachineLearningSamples-SentimentAnalysis"
        data-theme="medium"
      />
    </li>
    <li
      className="examplesitem item"
    >
      <div
        class="github-card"
        data-github="Azure/MachineLearningSamples-ChurnPrediction"
        data-theme="medium"
      />
    </li>
    <li
      className="examplesitem item"
    >
      <div
        class="github-card"
        data-github="Azure/MachineLearningSamples-BiomedicalEntityExtraction"
        data-theme="medium"
      />
    </li>
    <li
      className="examplesitem item"
    >
      <div
        class="github-card"
        data-github="Azure/MachineLearningSamples-QnAMatching"
        data-theme="medium"
      />
    </li>
    <li
      className="examplesitem item"
    >
      <div
        class="github-card"
        data-github="Azure/MachineLearningSamples-AerialImageClassification"
        data-theme="medium"
      />
    </li>
  </ul>
  <div
    class="paddles"
  >
    <button
      class="left-paddle paddle hidden"
      id="examples-left"
      onClick={[Function]}
    >
      &lt;
    </button>
    <button
      class="right-paddle paddle"
      id="examples-right"
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
</div>
`;

exports[`renders a snapshot of SliderContent (models) 1`] = `
<div>
  <ul
    className="menu models"
    id="models-menu"
  >
    <li
      className="modelsitem item"
    >
      <div
        class="github-card"
        data-github="Azure-Samples/AI-Gallery-InceptionV3"
        data-theme="medium"
      />
    </li>
    <li
      className="modelsitem item"
    >
      <div
        class="github-card"
        data-github="Azure-Samples/AI-Gallery-TinyYolo"
        data-theme="medium"
      />
    </li>
    <li
      className="modelsitem item"
    >
      <div
        class="github-card"
        data-github="onnx/models"
        data-theme="medium"
      />
    </li>
  </ul>
  <div
    class="paddles"
  >
    <button
      class="left-paddle paddle hidden"
      id="models-left"
      onClick={[Function]}
    >
      &lt;
    </button>
    <button
      class="right-paddle paddle"
      id="models-right"
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
</div>
`;
